shoes-core/lib/shoes/color/dsl.rb

```diff
@@ -0,0 +1,169 @@
+      [:lavenderblush, 255, 240, 245],
+      [:lawngreen, 124, 252, 0],
+      [:lemonchiffon, 255, 250, 205],
+      [:lightblue, 173, 216, 230],
+      [:lightcoral, 240, 128, 128],
+      [:lightcyan, 224, 255, 255],
+      [:lightgoldenrodyellow, 250, 250, 210],
+      [:lightgreen, 144, 238, 144],
+      [:lightgray, 211, 211, 211],
+      [:lightgrey, 211, 211, 211],
+      [:lightpink, 255, 182, 193],
+      [:lightsalmon, 255, 160, 122],
+      [:lightseagreen, 32, 178, 170],
+      [:lightskyblue, 135, 206, 250],
+      [:lightslategray, 119, 136, 153],
+      [:lightslategrey, 119, 136, 153],
+      [:lightsteelblue, 176, 196, 222],
+      [:lightyellow, 255, 255, 224],
+      [:lime, 0, 255, 0],
+      [:limegreen, 50, 205, 50],
+      [:linen, 250, 240, 230],
+      [:magenta, 255, 0, 255],
+      [:maroon, 128, 0, 0],
+      [:mediumaquamarine, 102, 205, 170],
+      [:mediumblue, 0, 0, 205],
+      [:mediumorchid, 186, 85, 211],
+      [:mediumpurple, 147, 112, 219],
+      [:mediumseagreen, 60, 179, 113],
+      [:mediumslateblue, 123, 104, 238],
+      [:mediumspringgreen, 0, 250, 154],
+      [:mediumturquoise, 72, 209, 204],
+      [:mediumvioletred, 199, 21, 133],
+      [:midnightblue, 25, 25, 112],
+      [:mintcream, 245, 255, 250],
+      [:mistyrose, 255, 228, 225],
+      [:moccasin, 255, 228, 181],
+      [:navajowhite, 255, 222, 173],
+      [:navy, 0, 0, 128],
+      [:oldlace, 253, 245, 230],
+      [:olive, 128, 128, 0],
+      [:olivedrab, 107, 142, 35],
+      [:orange, 255, 165, 0],
+      [:orangered, 255, 69, 0],
+      [:orchid, 218, 112, 214],
+      [:palegoldenrod, 238, 232, 170],
+      [:palegreen, 152, 251, 152],
+      [:paleturquoise, 175, 238, 238],
+      [:palevioletred, 219, 112, 147],
+      [:papayawhip, 255, 239, 213],
+      [:peachpuff, 255, 218, 185],
+      [:peru, 205, 133, 63],
+      [:pink, 255, 192, 203],
+      [:plum, 221, 160, 221],
+      [:powderblue, 176, 224, 230],
+      [:purple, 128, 0, 128],
+      [:red, 255, 0, 0],
+      [:rosybrown, 188, 143, 143],
+      [:royalblue, 65, 105, 225],
+      [:saddlebrown, 139, 69, 19],
+      [:salmon, 250, 128, 114],
+      [:sandybrown, 244, 164, 96],
+      [:seagreen, 46, 139, 87],
+      [:seashell, 255, 245, 238],
+      [:sienna, 160, 82, 45],
+      [:silver, 192, 192, 192],
+      [:skyblue, 135, 206, 235],
+      [:slateblue, 106, 90, 205],
+      [:slategray, 112, 128, 144],
+      [:slategrey, 112, 128, 144],
+      [:snow, 255, 250, 250],
+      [:springgreen, 0, 255, 127],
+      [:steelblue, 70, 130, 180],
+      [:tan, 210, 180, 140],
+      [:teal, 0, 128, 128],
+      [:thistle, 216, 191, 216],
+      [:tomato, 255, 99, 71],
+      [:turquoise, 64, 224, 208],
+      [:violet, 238, 130, 238],
+      [:wheat, 245, 222, 179],
+      [:white, 255, 255, 255],
+      [:whitesmoke, 245, 245, 245],
+      [:yellow, 255, 255, 0],
+      [:yellowgreen, 154, 205, 50],
+      [:shoes_background, 237, 237, 237],
+    ]
+
```

Comment:
I envisioned the system color being looked up somewhere around here (e.g. on load time of shoes, as the color should not change). Then we go to the backend one time and ask it for its background color and then add it to the list. I.e. something like (pseudo code!):

```ruby
system_background = @gui.system_color(:background) # especially uncertain here what to call/how to get it)
colors << [:system_background, system_background.red, system_background.green, system_background.blue]
```

That way we look up the system background color once on startup and from that point on it is just one normal color with the normal values assigned just like any other color.

The backend could even know the format (or maybe it should?) and we could do something like `app.gui.system_colors` or just `gui.system_colors` that then returns arrays like we'd expect them to. So like: `[[:system_background, red, green, blue], [:system_font, red2, green2, blue2]]`

Or the backend could have the responsibility of adding system colors so we'd do:

```
colors = some_backend_thing.add_system_colors(colors)
```

So many possibilities...